File: MeetingAssistant/app/src/main/java/com/nexsoft/meetingassistant/models/Laporan.kt
package com.nexsoft.meetingassistant.models

import com.google.gson.annotations.SerializedName

data class Laporan(
    @SerializedName("laporan_id") val laporanId: Int? = null,
    @SerializedName("hasil_id") val hasilId: Int? = null,
    @SerializedName("pr_id") val prId: Int? = null,
    @SerializedName("admin_id") val adminId: Int? = null,
    @SerializedName("file_laporan") val fileLaporan: String? = null,
    @SerializedName("tanggal_kirim") val tanggalKirim: String? = null
)

File: backend/app/schemas/laporan.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime, date


class LaporanCreate(BaseModel):
    """Schema for creating a new laporan."""
    hasil_id: int
    pr_id: Optional[int] = None
    admin_id: Optional[int] = None
    tanggal_kirim: Optional[datetime] = None


class LaporanResponse(BaseModel):
    """Schema for laporan response with relationships."""
    laporan_id: int
    hasil_id: Optional[int] = None
    pr_id: Optional[int] = None
    admin_id: Optional[int] = None
    file_laporan: Optional[str] = None
    tanggal_kirim: Optional[date] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    # Relationship data
    pemimpin_rapat_name: Optional[str] = None
    admin_name: Optional[str] = None
    nama_rekaman: Optional[str] = None

    class Config:
        from_attributes = True
